bcipy/gui/mode/RSVPKeyboard.py

```diff
@@ -1,210 +1,281 @@
-                           wildcard='JSON files (*.json)|*.json',
-                           style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST) as fd:
-            if fd.ShowModal() != wx.ID_CANCEL:
-                self.parameter_location = fd.GetPath()
-
-    def edit_parameters(self, _event) -> None:
-        """Edit Parameters. Prompts for a parameters.json file to use. If the default parameters
-        are selected a copy is used.
+        self.parameters = load_json_parameters(
+            self.parameter_location, value_cast=True)
+
+        self.task_colors = itertools.cycle(
+            ['blue', 'green', 'yellow', 'red', 'limegreen', 'gray', 'pink'])
+
+    def select_parameters(self) -> None:
+        """Select Parameters.
+
+        Opens a dialog to select the parameters.json configuration to use.
+        """
+
+        response = self.get_filename_dialog(message='Select parameters file', file_type='JSON files (*.json)')
+        if response:
+            self.parameter_location = response
+
+    def edit_parameters(self) -> None:
+        """Edit Parameters.
+
+        Prompts for a parameters.json file to use. If the default parameters are selected, a copy is used.
         """
         if self.parameter_location == DEFAULT_PARAMETERS_PATH:
             # Don't allow the user to overwrite the defaults
-            with wx.MessageDialog(
-                    self,
-                    "The default parameters.json can't be overridden. A copy will be used.",
-                    'Info', wx.OK | wx.CANCEL) as confirm_dialog:
-                response = confirm_dialog.ShowModal()
-                if response == wx.ID_OK:
-                    self.parameter_location = copy_parameters()
-                else:
-                    return
+            response = self.throw_alert_message(
+                title='BciPy Alert',
+                message='The default parameters.json cannot be overridden. A copy will be used.',
+                message_type=AlertMessageType.INFO,
+                okay_or_cancel=True)
+
+            if response == AlertResponse.OK.value:
+                self.parameter_location = copy_parameters()
+            else:
+                return None
 
         subprocess.call(
             f'python bcipy/gui/params_form.py -p {self.parameter_location}',
             shell=True)
 
-    def launch_bci_main(self, event: wx.Event) -> None:
-        """Launch BCI MAIN"""
+    def launch_bci_main(self) -> None:
+        """Launch BCI main.
+
+        If a user id has been selected, get the id set on the task button and execute
+            the task in a new shell.
+        """
         if self.check_input():
             self.event_started = True
-            username = self.comboboxes[0].GetValue().replace(" ", "_")
-            experiment_type = event.GetEventObject().GetId()
+            username = self.user_input.currentText()
+            experiment_type = self.sender().get_id()
```

Comment:
It took a bit of digging to figure out how `self.user_input` is initialized. For clarity, we might want to declare this property in the `__init__` and add a comment that it gets set in the `show_gui` method.